### PR TITLE
feat(nix): flakify

### PR DESCRIPTION
**Pull request by MangoIV**
###### Things done

- [x] Tested, as applicable:
  - [x] Manually
- [x] Updated [CHANGELOG.md](../CHANGELOG.md) (if applicable).
- [x] Fits [CONTRIBUTING.md](https://github.com/MrcJkb/haskell-tools.nvim/blob/master/CONTRIBUTING.md)

This is still in draft format as I want to add something more so one can properly use that within their system configuration.

### CODE REVIEW

**Comment by MangoIV** [comment]
For some reason I cannot get it to work, it's not the fault of this nix code, though, as I also cannot get it to work with the nixpkgs version; for some reason I get undefined token repl when trying to use the repl options and the lsp doesn't register; i.e. all of the language servers show up but the haskell one. What could be ther reason?

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ### Added
 - Automatically generate project & package tags if [`fast-tags`](https://github.com/elaforge/fast-tags) is installed.
 - Configuration for falling back to hoogle search if 'vim.lsp.definition' fails.
+- flake setup
```
Comment (MrcJkb) [inline CHANGELOG.md:11]:
```suggestion
- Nix flake setup
```

File: README.md [hunk 2]
```diff
@@ -56,6 +56,8 @@ use {
 [^1] It is suggested to use the [latest release tag](./CHANGELOG.md)
      and to update it manually, if you would like to avoid breaking changes.
 
+This project provides flake outputs in the form of a package and an overlay; use it as you wish in your nixos or 
+home-manager configuration.
```
Comment (MrcJkb) [inline README.md:60]:
```suggestion
For NixOS users with flakes enabled, this project provides outputs in the form of a package and an overlay; use it as you wish in your nixos or home-manager configuration.
It is also available on `nixpkgs`.
```

**Comment by MrcJkb** [comment]
Thanks so much for this! Now I finally know how to use GitHub actions with flakes :tada:

**Review by MrcJkb — CHANGES_REQUESTED** [review]
Just some minor nitpicks...

**Review by MrcJkb — APPROVED** [review]

**Comment by MangoIV** [comment]
Thank you!

**Pull request merged**